test/integration/etcd/etcd_storage_path_test.go

```diff
@@ -0,0 +1,933 @@
+						t.Fatalf("failed to clean up etcd: %#v", err)
+					}
+				}
+			}()
+
+			if err := client.createPrerequisites(mapper, testNamespace, testData.prerequisites, all); err != nil {
+				t.Errorf("failed to create prerequisites for %s from %s: %#v", kind, pkgPath, err)
+				return
+			}
+
+			if shouldCreate { // do not try to create items with no stub
+				if err := client.create(testData.stub, testNamespace, mapping, all); err != nil {
+					t.Errorf("failed to create stub for %s from %s: %#v", kind, pkgPath, err)
+					return
+				}
+			}
+
+			output, err := getFromEtcd(kvClient, testData.expectedEtcdPath)
+			if err != nil {
+				t.Errorf("failed to get from etcd for %s from %s: %#v", kind, pkgPath, err)
+				return
+			}
+
+			expectedGVK := gvk
+			if testData.expectedGVK != nil {
+				if gvk == *testData.expectedGVK {
+					t.Errorf("GVK override %s for %s from %s is unnecessary or something was changed incorrectly", testData.expectedGVK, kind, pkgPath)
+				}
+				expectedGVK = *testData.expectedGVK
+			}
+
+			actualGVK := output.getGVK()
+			if actualGVK != expectedGVK {
+				t.Errorf("GVK for %s from %s does not match, expected %s got %s", kind, pkgPath, expectedGVK, actualGVK)
+			}
+
+			if !apiequality.Semantic.DeepDerivative(input, output) {
+				t.Errorf("Test stub for %s from %s does not match: %s", kind, pkgPath, diff.ObjectGoPrintDiff(input, output))
+			}
+		}()
+	}
+
+	if inEtcdData, inEtcdSeen := diffMaps(etcdStorageData, etcdSeen); len(inEtcdData) != 0 || len(inEtcdSeen) != 0 {
+		t.Errorf("etcd data does not match the types we saw:\nin etcd data but not seen:\n%s\nseen but not in etcd data:\n%s", inEtcdData, inEtcdSeen)
+	}
+
+	if inEphemeralWhiteList, inEphemeralSeen := diffMaps(ephemeralWhiteList, ephemeralSeen); len(inEphemeralWhiteList) != 0 || len(inEphemeralSeen) != 0 {
+		t.Errorf("ephemeral whitelist does not match the types we saw:\nin ephemeral whitelist but not seen:\n%s\nseen but not in ephemeral whitelist:\n%s", inEphemeralWhiteList, inEphemeralSeen)
+	}
+
+	if inKindData, inKindSeen := diffMaps(kindWhiteList, kindSeen); len(inKindData) != 0 || len(inKindSeen) != 0 {
+		t.Errorf("kind whitelist data does not match the types we saw:\nin kind whitelist but not seen:\n%s\nseen but not in kind whitelist:\n%s", inKindData, inKindSeen)
+	}
+}
+
+func startRealMasterOrDie(t *testing.T) (*allClient, clientv3.KV, meta.RESTMapper) {
+	_, defaultServiceClusterIPRange, err := net.ParseCIDR("10.0.0.0/24")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeClientConfigValue := atomic.Value{}
+	storageConfigValue := atomic.Value{}
+
+	go func() {
+		for {
+			kubeAPIServerOptions := options.NewServerRunOptions()
+			kubeAPIServerOptions.SecureServing.BindAddress = net.ParseIP("127.0.0.1")
+			kubeAPIServerOptions.Etcd.StorageConfig.ServerList = []string{framework.GetEtcdURLFromEnv()}
+			kubeAPIServerOptions.Etcd.DefaultStorageMediaType = runtime.ContentTypeJSON // TODO use protobuf?
```

Comment:
Sure, but you have to tell me how to extract protobuf from etcd without using `kapi.Codecs`.

Basically, I want to make sure no conversions, defaulting, etc happens.